pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -532,18 +545,81 @@ func createNodeSecurityGroup(client *gophercloud.ServiceClient, nodeSecurityGrou
 	return nil
 }
 
+//getStringFromServiceAnnotation searches a given v1.Service for a specific annotationKey and either returns the annotation's value or a specified defaultSetting
+func getStringFromServiceAnnotation(service *v1.Service, annotationKey string, defaultSetting string, enabledFlag bool) string {
+	glog.V(4).Infof("getStringFromServiceAnnotation(%v, %v, %v, %v)", service, annotationKey, defaultSetting, enabledFlag)
+	if !enabledFlag {
+		return defaultSetting
+	}
+	if annotationValue, ok := service.Annotations[annotationKey]; ok {
+		//if there is an annotation for this setting, set the "setting" var to it
+		glog.V(4).Infof("Found a Service Annotation: %v = %v", annotationKey, annotationValue)
+		return annotationValue
+	}
+	//if there is no annotation, set "settings" var to the value from cloud config
+	glog.V(4).Infof("Could not find a Service Annotation; falling back on cloud-config setting: %v = %v", annotationKey, defaultSetting)
+	return defaultSetting
+}
+
+//getBoolFromServiceAnnotation searches a given v1.Service for a specific annotationKey and either returns the annotation's value or a specified defaultSetting
```

Comment:
There's a reasonable amount of duplicated code here and in the Duration version below.  I think I'd rather we just called `getStringFromServiceAnnotation` to do the heavy lifting in each of these, and then parsed the (string) result.  I don't care whether you take the default value in string form always, or take a properly typed value and then convert it to a string (somewhat unnecessarily) for `getStringFromServiceAnnotation`.